public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useStyles2 } from '@grafana/ui';
+import { Messages } from 'app/percona/add-instance/components/AddRemoteInstance/FormParts/FormParts.messages';
+import { getStyles } from 'app/percona/add-instance/components/AddRemoteInstance/FormParts/FormParts.styles';
+import { agentId } from 'app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.constants';
+import { NodesAgentsProps } from 'app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.types';
+import { GET_NODES_CANCEL_TOKEN } from 'app/percona/inventory/Inventory.constants';
+import { AgentsOption, NodesOption } from 'app/percona/inventory/Inventory.types';
+import { SelectField } from 'app/percona/shared/components/Form/SelectFieldCore';
+import { useCancelToken } from 'app/percona/shared/components/hooks/cancelToken.hook';
+import { nodesOptionsMapper } from 'app/percona/shared/core/reducers/nodes';
+import { fetchNodesAction } from 'app/percona/shared/core/reducers/nodes/nodes';
+import { getNodes } from 'app/percona/shared/core/selectors';
+import { isApiCancelError } from 'app/percona/shared/helpers/api';
+import { logger } from 'app/percona/shared/helpers/logger';
+import { useAppDispatch } from 'app/store/store';
+import { useSelector } from 'app/types';
+
+export const NodesAgents: FC<NodesAgentsProps> = ({ form }) => {
+  const styles = useStyles2(getStyles);
+  const dispatch = useAppDispatch();
+  const [generateToken] = useCancelToken();
+  const [selectedNode, setSelectedNode] = useState<NodesOption>();
+  const [selectedAgent, setSelectedAgent] = useState<AgentsOption>();
+  const { nodes } = useSelector(getNodes);
+
+  const nodesOptions = useMemo<NodesOption[]>(() => nodesOptionsMapper(nodes), [nodes]);
+
+  const loadData = useCallback(async () => {
+    try {
+      await dispatch(fetchNodesAction({ token: generateToken(GET_NODES_CANCEL_TOKEN) })).unwrap();
+    } catch (e) {
+      if (isApiCancelError(e)) {
+        return;
+      }
+      logger.error(e);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const changeAgentValue = (value: AgentsOption) => {
+    setSelectedAgent(value);
+
+    if (value.label !== agentId.pmmServer) {
+      form?.change('address', 'localhost');
+    }
+  };
+
+  const setNodeAndAgent = (value: NodesOption) => {
+    setSelectedNode(value);
+
+    if (value.agents && value.agents.length > 1) {
+      const pmmServerAgent = value.agents.find((item) => item.value === agentId.pmmServer);
+      if (pmmServerAgent) {
+        form?.change('pmm_agent_id', pmmServerAgent);
+      }
+    } else if (value.agents && value.agents.length === 1) {
+      form?.change('pmm_agent_id', value.agents[0]);
+    }
+
+    if (selectedAgent && selectedAgent.label !== agentId.pmmServer) {
```

Comment:
I've checked, and this condition will not work as expected, because selectedAgent, will be set to the **previously** selected agent. (First time it runs it is undefined then the value that was selected before)

What you can do is store the value of the agent that is selected in the form:

```suggestion
let selectedAgent: AgentsOption;

if (value.agents?.length > 1) {
  selectedAgent = value.agents.find((item) => item.value === agentId.pmmServer);
} else if (value.agents?.length === 1) {
  selectedAgent = value.agents[0];
}

if (selectedAgent) {
  form?.change('pmm_agent_id', selectedAgent);
  
  if (selectedAgent.value !== PMM_SERVER_NODE_AGENT_ID) {
    form?.change('address', 'localhost');
  }
}
```

Reply:
added this code